File: src/main/java/com/skyspacefourtyfour/service/BaggageClaimService.java
package com.skyspacefourtyfour.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefourtyfour.domain.BaggageClaim;
import com.skyspacefourtyfour.dto.BaggageClaimDTO;
import com.skyspacefourtyfour.dto.BaggageClaimSearchDTO;
import com.skyspacefourtyfour.dto.BaggageClaimPageDTO;
import com.skyspacefourtyfour.dto.BaggageClaimConvertCriteriaDTO;
import com.skyspacefourtyfour.service.GenericService;
import com.skyspacefourtyfour.dto.common.RequestDTO;
import com.skyspacefourtyfour.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface BaggageClaimService extends GenericService<BaggageClaim, Integer> {

	List<BaggageClaim> findAll();

	ResultDTO addBaggageClaim(BaggageClaimDTO baggageClaimDTO, RequestDTO requestDTO);

	ResultDTO updateBaggageClaim(BaggageClaimDTO baggageClaimDTO, RequestDTO requestDTO);

    Page<BaggageClaim> getAllBaggageClaims(Pageable pageable);

    Page<BaggageClaim> getAllBaggageClaims(Specification<BaggageClaim> spec, Pageable pageable);

	ResponseEntity<BaggageClaimPageDTO> getBaggageClaims(BaggageClaimSearchDTO baggageClaimSearchDTO);
	
	List<BaggageClaimDTO> convertBaggageClaimsToBaggageClaimDTOs(List<BaggageClaim> baggageClaims, BaggageClaimConvertCriteriaDTO convertCriteria);

	BaggageClaimDTO getBaggageClaimDTOById(Integer baggageClaimId);







}

File: skyspacefourtyfour_ui/src/services/BaggageClaimService.js
import http from "../http-common"; 

class BaggageClaimService {
  getAllBaggageClaims(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/baggageClaim/baggageClaims`, searchDTO);
  }

  get(baggageClaimId) {
    return this.getRequest(`/baggageClaim/${baggageClaimId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/baggageClaim?field=${matchData}`, null);
  }

  addBaggageClaim(data) {
    return http.post("/baggageClaim/addBaggageClaim", data);
  }

  update(data) {
  	return http.post("/baggageClaim/updateBaggageClaim", data);
  }
  
  uploadImage(data,baggageClaimId) {
  	return http.postForm("/baggageClaim/uploadImage/"+baggageClaimId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new BaggageClaimService();
